kaldo/observables/secondorder.py

```diff
@@ -9,11 +9,190 @@
 from kaldo.controllers.displacement import calculate_second
 import ase.units as units
 from kaldo.helpers.logger import get_logger, log_size
+from pathlib import Path # NEW !
+from ase.geometry import get_distances # NEW !
+import sys # NEW !
 
 logging = get_logger()
 
 SECOND_ORDER_FILE = 'second.npy'
 
+def parse_tdep_forceconstant(
+        fc_file="infile.forceconstants",
+        primitive="infile.ucposcar",
+        supercell="infile.ssposcar",
+        fortran=True,
+        two_dim=True,
+        symmetrize=False,
+        reduce_fc=True,
+        eps=1e-13,
+        tol=1e-5,
+        format="vasp", ):
+    
+    if isinstance(primitive, Atoms):
+        uc = primitive
+    elif Path(primitive).exists():
+        uc = ase.io.read(primitive, format=format)
+    else:
+        raise RuntimeError("primitive cell missing")
+
+    if isinstance(supercell, Atoms):
+        sc = supercell
+    elif Path(supercell).exists():
+        sc = ase.io.read(supercell, format=format)
+    else:
+        raise RuntimeError("supercell missing")
+
+    uc.wrap(eps=tol)
+    sc.wrap(eps=tol)
+    n_uc = len(uc)
+    n_sc = len(sc)
+
+    force_constants = np.zeros((len(uc), len(sc), 3, 3))
+
+    with open(fc_file) as fo:
+        n_atoms = int(next(fo).split()[0])
+        cutoff = float(next(fo).split()[0])
+
+        assert n_atoms == n_uc, f"n_atoms == {n_atoms}, should be {n_uc}"
+
+        #print(f".. Number of atoms:   {n_atoms}")
+        #print(rf".. Real space cutoff: {cutoff:.3f} \AA")
+
+        for i1 in range(n_atoms):
+            n_neighbors = int(next(fo).split()[0])
+            for _ in range(n_neighbors):
+                i2 = int(next(fo).split()[0]) - 1
+                lp = np.array(next(fo).split(), dtype=float)
+                phi = np.array([next(fo).split() for _ in range(3)], dtype=float)
+                r_target = uc.positions[i2] + np.dot(lp, uc.cell[:])
+                for ii, r1 in enumerate(sc.positions):
+                    r_diff = np.abs(r_target - r1)
+                    sc_temp = sc.get_cell(complete=True)
+                    r_diff = np.linalg.solve(sc_temp.T, r_diff.T).T % 1.0
+                    r_diff -= np.floor(r_diff + eps)
+                    if np.sum(r_diff) < tol:
+                        force_constants[i1, ii, :, :] += phi
+
+    if not reduce_fc or two_dim:
+        force_constants = remap_force_constants(
+            force_constants, uc, sc, symmetrize=symmetrize
+        )
+
+    if two_dim:
+        return force_constants.swapaxes(2, 1).reshape(2 * (3 * n_sc,))
```

Comment:
Is `reshape(2 * (3 * n_sc,))` correct? Should that be `reshape(2 * (3 * n_sc))`?